### PR TITLE
RELATED: RAIL-3428, RAIL-3840 plugin-related improvements

### PR DESCRIPTION
**Pull request by no23reason**
* RAIL-3428 - Target ES2017 in dashboard plugins. This is more forward compatible with gdc-dashboards dropping IE support: the current gdc-dashboards can run ES2017 plugins and gdc-dashboards without IE support can run them too. They can't run ES5 plugins, though.
* RAIL-3840 -  improve decorator typings to prevent undefined results of the providerFactory.

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/plugins/customizer.ts [hunk 1]
```diff
@@ -135,6 +138,9 @@ export interface IDashboardKpiCustomizer {
      *
      *         function MyCustomDecorator() {
      *              const Decorated = next(kpi, widget);
+     *              if (!Decorated) {
+     *                  return null;
+     *              }
```
Comment (lupko) [inline libs/sdk-ui-dashboard/src/plugins/customizer.ts:143]: hmm, ins't there a problem with typing of next(); decorator should always work on top of something, the underlying infra should ensure that.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5808/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7571/ : SUCCESS in 17m 04s

File: libs/sdk-ui-dashboard/src/plugins/customizer.ts (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/plugins/customizer.ts]: Fixed by improving the typings in the area

**Review by plavemar — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1920

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1665/ : SUCCESS in 2m 55s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1945/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/742/ : SUCCESS in 18m 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7574/ : SUCCESS in 18m 39s

**Pull request merged**